src/WebJobs.Script/Diagnostics/Extensions/LoggerExtension.cs

```diff
@@ -206,6 +206,11 @@ internal static class LoggerExtension
         private static readonly Action<ILogger, string, Exception> _publishingMetrics =
             LoggerMessage.Define<string>(LogLevel.Debug, new EventId(338, nameof(PublishingMetrics)), "{metrics}");
 
+        private static readonly Action<ILogger, string, Exception> _outdatedExtensionBundle =
+           LoggerMessage.Define<string>(LogLevel.Warning,
+           new EventId(342, nameof(OutdatedExtensionBundle)),
+           "{message}");
```

Comment:
This is not using high perf logging correctly. The point is to have the template defined here and only fill in the non-constant parts at runtime. (Also`_publishingMetrics` is a bad example, it shouldn't be doing that either).

I see you have 2 different templates, what I suggest is:

1. Define two different templates via `LoggerMessage.Define`.
2. Define a _single_ extension method on `ILogger` for this. Move all of the templating logic into this message. But you will just be trying to get the individual values to pass to the messages defined in step 1.
3. Make sure to check `logger.IsEnabled(LogLevel.Warning);` before doing any calculations so you can no-op as an optimization.